conftest.py

```diff
@@ -168,6 +195,13 @@ def kill_all_test_spawned_processes() -> None:
         print("Not all processes were killed after test")
 
 
+@pytest.fixture
+def mpi_app_path() -> str:
```

Comment:
Looks like we are trying to run this fixture exactly once per test suite run. Can we set the scope of this fixture to `"session"`?
```suggestion
@pytest.fixture(scope="session")
def mpi_app_path() -> str:
```

Reply:
Seems reasonable